libs/wingc/src/type_check.rs

```diff
@@ -3388,6 +3298,106 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn resolve_variable_from_instance_type(
```

Comment:
My cognative ability to name things is out of service today 😅 I needed to extract this logic to a function to recursively call it reference is `Type::Optional` I would have liked to do maybe a closure instead but to make closures recursive in rust was hella ugly.